server/container/core/src/test/java/org/apache/james/server/core/MimeMessageCopyOnWriteProxyTest.java

```diff
@@ -245,6 +247,18 @@ public void testMessageCloningViaCoW3() throws Exception {
         LifecycleUtil.dispose(mm);
     }
 
+    @Test
+    public void testMessageWithWrongContentTypeShouldNotThrow() throws Exception {
+        ImmutableList<MailAddress> recipients = ImmutableList.of(new MailAddress("recipient@test.com"));
+        MimeMessageCopyOnWriteProxy messageFromSources = (MimeMessageCopyOnWriteProxy) getMessageFromSources(
+                content + sep + body);
+        MailImpl mail = new MailImpl("test", new MailAddress("test@test.com"), recipients, messageFromSources);
+        mail.getMessage().addHeader("Content-Type", "file;name=\"malformed.pdf\"");
+        mail.getMessage().saveChanges();
```

Comment:
As far as I understand, the bug is not specific to MimeMessageCopyOnWriteProxy but is common to MimeMessage, don't you think ?

Reply:
Maybe we can delegate that responsibility in a Guice ConfigurationPerformer to set the static properties for Javax.mail system wide?

Reply:
I like that, thanks